src/engine.go

```diff
@@ -161,19 +161,30 @@ func (e *engine) print() string {
 		if !*e.env.getArgs().Eval {
 			break
 		}
-		// escape double quotes contained in the prompt
-		prompt := fmt.Sprintf("PS1=\"%s\"", strings.ReplaceAll(e.string(), "\"", "\"\""))
-		prompt += fmt.Sprintf("\nRPROMPT=\"%s\"", e.rprompt)
+		// escape double quotes contained in the left prompt
+		lprompt := strings.ReplaceAll(e.string(), "\"", "\"\"")
```

Comment:
Looking at how these lines are identical, it would make sense to create a helper function for this. That way extending is a lot more straightforward. Maybe `escapePrompt(prompt string) string`?